linker/src/lib.rs

```diff
@@ -93,31 +93,32 @@ pub fn link<T: FieldElement>(graph: PILGraph<T>) -> Result<PILFile<T>, Vec<Strin
             let to = link.to;
 
             // the lhs is `instr_flag { inputs, outputs }`
+            let mut expressions =
+                Vec::with_capacity(1 + from.params.inputs.len() + from.params.outputs.len());
+            //flag
+            expressions.extend(to.operation.id.iter().cloned().map(Expression::Number));
+            // inputs
+            expressions.extend(
+                from.params
+                    .inputs
+                    .into_iter()
+                    .map(|i| {
+                        assert!(i.ty.is_none() || i.ty == Some("write".to_string()));
```

Comment:
isn't there something in the type's type that tell us this, so we don't need to use strings?